workspaces/rbac/plugins/rbac/src/components/CreateRole/MemberSelectionDialog.tsx

```diff
@@ -0,0 +1,305 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect, useMemo } from 'react';
+import {
+  GroupEntity,
+  stringifyEntityRef,
+  UserEntity,
+} from '@backstage/catalog-model';
+import { Table, TableColumn, Progress } from '@backstage/core-components';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Tabs,
+  Tab,
+  Box,
+  makeStyles,
+  createStyles,
+  Theme,
+  IconButton,
+} from '@material-ui/core';
+import CloseIcon from '@mui/icons-material/Close';
+import { TranslationFunction } from '@backstage/core-plugin-api/alpha';
+import { rbacTranslationRef } from '../../translations';
+import { SelectedMember } from './types';
+import { MemberEntity } from '../../types';
+import {
+  getChildGroupsCount,
+  getMembersCount,
+  getParentGroupsCount,
+} from '../../utils/create-role-utils';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    dialogCloseButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+  }),
+);
+
+const getDescription = (
+  member: MemberEntity,
+  t: TranslationFunction<typeof rbacTranslationRef.T>,
+) => {
+  const memberCount = getMembersCount(member);
+  const parentCount = getParentGroupsCount(member);
+  const childCount = getChildGroupsCount(member);
+
+  return member.kind === 'Group'
+    ? [
+        memberCount > 0
+          ? t('common.membersCount' as any, { count: memberCount })
+          : '',
+        parentCount > 0
+          ? t('common.parentGroupCount' as any, { count: parentCount })
+          : '',
+        childCount > 0
+          ? t('common.childGroupsCount' as any, { count: childCount })
+          : '',
+      ]
+        .filter(Boolean)
+        .join(', ')
+    : undefined;
+};
+
+const entityToSelectedMember = (
+  entity: MemberEntity,
+  t: TranslationFunction<typeof rbacTranslationRef.T>,
+): SelectedMember => {
+  const ref = stringifyEntityRef(entity);
+  return {
+    id: entity.metadata.etag ?? ref,
+    label: entity.spec?.profile?.displayName ?? entity.metadata.name,
+    description: getDescription(entity, t),
+    etag: entity.metadata.etag ?? ref,
+    type: entity.kind,
+    namespace: entity.metadata.namespace,
+    members: getMembersCount(entity),
+    ref: ref,
+  };
+};
+
+function TabPanel(props: {
+  children?: React.ReactNode;
+  index: number;
+  value: number;
+}) {
+  const { children, value, index, ...other } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`simple-tabpanel-${index}`}
+      aria-labelledby={`simple-tab-${index}`}
+      {...other}
+    >
+      {value === index && <Box sx={{ pt: 2 }}>{children}</Box>}
+    </div>
+  );
+}
+
+const userColumns: TableColumn<UserEntity>[] = [
+  { title: 'Name', field: 'spec.profile.displayName' },
```

Comment:
What happens if they don't have a `displayName` and same on line 129? The examples you include they all do but not all Backstage Adopters may follow that strictly.